dataprofiler/tests/profilers/test_json_encoder.py

```diff
@@ -29,3 +31,60 @@ def test_encode_base_column_profiler(self):
         )
 
         self.assertEqual(json.loads(serialized), exepcted)
+
+    def test_encode_categorical_column_profiler(self):
+        profile = CategoricalColumn("0")
+
+        serialized = json.dumps(profile, cls=ProfileEncoder)
+        expected = json.dumps(
+            {
+                "name": "0",
+                "col_index": np.nan,
+                "sample_size": 0,
+                "metadata": dict(),
+                "times": defaultdict(),
+                "thread_safe": True,
+                "_categories": defaultdict(int),
+                "_CategoricalColumn__calculations": dict(),
+                "_top_k_categories": None,
+            }
+        )
+
+        self.assertEqual(json.loads(serialized), json.loads(expected))
+
+    def test_encode_categorical_column_profiler_after_update(self):
+        df_categorical = pd.Series(
+            [
+                "a",
+                "a",
+                "a",
+                "b",
+                "b",
+                "b",
+                "b",
+                "c",
+                "c",
+                "c",
+                "c",
+                "c",
+            ]
+        )
+        profile = CategoricalColumn(df_categorical.name)
+        profile.update(df_categorical)
+
+        serialized = json.dumps(profile, cls=ProfileEncoder)
+        exepcted = json.dumps(
+            {
+                "name": profile.name,
+                "col_index": profile.col_index,
+                "sample_size": profile.sample_size,
+                "metadata": profile.metadata,
+                "times": profile.times,
+                "thread_safe": profile.thread_safe,
+                "_categories": profile._categories,
+                "_CategoricalColumn__calculations": {},
+                "_top_k_categories": profile._top_k_categories,
+            }
+        )
```

Comment:
Yeah instead of using the `profile.` these values should be literals. This is to prevent changes from going under the radar and forcing the test to have specific values, rather than change with your profile object.

Reply:
same as my comment above